src/main/scala/firrtl/passes/AnnotateMemMacros.scala

```diff
@@ -0,0 +1,164 @@
+package firrtl.passes
+
+import scala.collection.mutable.{ArrayBuffer,HashMap}
+import AnalysisUtils._
+import firrtl.WrappedExpression._
+import firrtl.ir._
+import firrtl._
+import firrtl.Utils._
+
+case class AppendableInfo(fields: Map[String,Any]) extends Info {
+  def append(a: Map[String,Any]) = this.copy(fields = fields ++ a)
+  def append(a: (String,Any)): AppendableInfo = append(Map(a))
+  def get(f: String) = fields.get(f)
+  override def equals(b: Any) = b match {
+    case i: AppendableInfo => fields - "info" == i.fields - "info"
+    case _ => false
+  }
+}
+
+object AnalysisUtils {
+
+  def getConnects(m: Module) = {
+    val connects = HashMap[String, Expression]()
+    def getConnects(s: Statement): Unit = s match {
+      case s: Connect =>
+        connects(s.loc.serialize) = s.expr
+      case s: PartialConnect =>
```

Comment:
FYI -- this was also taken more or less from InferReadWrite.scala. I think these are useful functions that should be pulled out into some generic AnalysisUtils-like Util thing